### PR TITLE
Make sure size=0 works on the inner_hits level.

### PR DESCRIPTION
**Pull request by martijnvg**
PR for #10358

### CODE REVIEW

File: src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java [hunk 1]
```diff
@@ -135,8 +123,26 @@ public TopDocs topDocs(SearchContext context, FetchSubPhase.HitContext hitContex
             BitDocIdSetFilter parentFilter = context.bitsetFilterCache().getBitDocIdSetFilter(rawParentFilter);
             Filter childFilter = context.filterCache().cache(childObjectMapper.nestedTypeFilter(), null, context.queryParserService().autoFilterCachePolicy());
             Query q = new FilteredQuery(query, new NestedChildrenFilter(parentFilter, childFilter, hitContext));
-            context.searcher().search(q, topDocsCollector);
-            return topDocsCollector.topDocs(from(), size());
+
+            int topN = from() + size();
+            if (topN == 0) {
```
Comment (jpountz) [inline src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java]: I think you should just check size() == 0? If from is 3 and size 0, we still only need to count matches?

File: src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java [hunk 2]
```diff
@@ -281,11 +279,28 @@ public TopDocs topDocs(SearchContext context, FetchSubPhase.HitContext hitContex
...
             Filter filter = new TermFilter(new Term(field, term)); // Only include docs that have the current hit as parent
             Filter typeFilter = documentMapper.typeFilter(); // Only include docs that have this inner hits type.
-            context.searcher().search(
-                    new FilteredQuery(query, new AndFilter(Arrays.asList(filter, typeFilter))),
-                    topDocsCollector
-            );
-            return topDocsCollector.topDocs(from(), size());
+
+            int topN = from() + size();
+            if (topN == 0) {
```
Comment (jpountz) [inline src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java]: same here

**Comment by jpountz** [comment]
Left one minor comment, otherwise LGTM. Feel free to push without further review!

File: src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java (hunk 1 above)
Comment (martijnvg) [inline src/main/java/org/elasticsearch/search/fetch/innerhits/InnerHitsContext.java]: good point, I'll change that

**Pull request merged**